File: src/utils/Winning.scala
package utils

import battlecode.common.GameActionException
import utils.Constants.BULLETS_TO_WIN
import utils.Current.I

/**
  * Winning utilities.
  */
object Winning {

  /**
    * If all required bullets are there, donates them all and wins the game.
    */
  def tryInstantWin(): Unit =
    try {
      if (I.getTeamBullets >= BULLETS_TO_WIN)
        I donate BULLETS_TO_WIN
    }
    catch { case _: GameActionException => }

}

File: src/utils/Winning.java
package utils;

import battlecode.common.GameActionException;

import static utils.Constants.BULLETS_TO_WIN;
import static utils.Current.*;

/**
 * Winning utilities.
 */
public class Winning {

    public static void tryInstantWin() {
        if (I.getTeamBullets() >= BULLETS_TO_WIN)
            try {
                I.donate(BULLETS_TO_WIN);
            } catch (GameActionException e) {
                System.out.println("Cannot donate");
                e.printStackTrace();
            }
    }
}
